aiida_optimize/engines/_bisection.py

```diff
@@ -93,19 +93,23 @@ def _update(self, outputs: ty.Dict[int, ty.Any]) -> None:
             else:
                 self.lower = self.average
 
-    def _get_optimal_result(self) -> ty.Tuple[int, Result]:
+    def _get_optimal_result(self) -> ty.Tuple[int, ty.Any, Result]:
         """
         Return the index and optimization value of the best evaluation workflow.
         """
         output_values = {
             key: get_nested_result(value.output, self.result_key)
             for key, value in self._result_mapping.items()
         }
-        return min(output_values.items(), key=lambda item: abs(item[1].value - self.target_value))
+        opt_index, opt_output = min(
+            output_values.items(), key=lambda item: abs(item[1].value - self.target_value)
+        )
+        opt_input = self._result_mapping[opt_index].input[self.input_key]
+        return (opt_index, opt_input, opt_output)
 
     @property
     def result_value(self) -> Result:
```

Comment:
I think this method (and `result_index`) can be removed here, it's implemented in the base class.